### ISSUE TITLE
Replacing ProductGridContainer via Components.API throws error.

### ISSUE THREAD

**Issue by prinzdezibel** [reactioncommerce/reaction#3382]
Trying to replace ProductGridContainer will result in Javascript errors when rendering ProductGrid (LandingPage), because `this.state.productIds` of ProductGridContainer is not defined. The page will not load, but render the spinner indefinitely.

### Expected behavior
Overriding of ProductGridContainer via `replaceComponent` should work without errors.

### Actual behavior
Error is thrown. Spinner is shown  forever.

### Steps to reproduce the behavior

Somewhere in code: 
```
import { replaceComponent, getRawComponent, getHOCs } from "@reactioncommerce/reaction-components";

replaceComponent("ProductGrid", getRawComponent("ProductGrid"), getHOCs("ProductGrid"));
```

Navigate to `http://localhost:3000`

See errors in browser console.

### Versions
```
Node: 8.2.1
NPM: 5.3.0
Meteor Node: 8.8.1
Meteor NPM: 5.4.2
Reaction CLI: 0.23.1
Reaction: 1.6.0
Reaction branch: master
Docker: 17.03.1-ce
```

**Comment by jshimko** [comment]
Is there a real world use case for your above example?  It's replacing the `ProductGrid` component with itself and then re-wrapping it with the same HOC's.

**Comment by prinzdezibel** [comment]
thats only a show case. i m doing it with another component thats really overridden. the result is the same, though

**Comment by prinzdezibel** [comment]
@jshimko You were right. PEBCAB. The "re-wrapping part" of your question was the culprit.

Stupid me. I've read the Components.API docs, but obviously not close enough, which is why I did:
I was not aware that the original HOCs are already mixed into when replacing a component. I've read about the Component.API in our docs, but obviously not close enough, which is why I did:

```
replaceComponent("ProductGrid", MyProductGrid, getHOCs("ProductGrid"));
```
instead of
```
replaceComponent("ProductGrid", MyProductGrid);
```

I was not aware that the original HOCs gets merged in automatically when calling `replaceComponent`.

Sorry for wasting your time :(

Going to close this issue.
